runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -9295,6 +9292,9 @@ J9::Z::TreeEvaluator::VMmonentEvaluator(TR::Node * node, TR::CodeGenerator * cg)
          generateS390LabelInstruction(cg, TR::InstOpCode::label, node, helperCallLabel );
          TR::RegisterDependencyConditions *deps = NULL;
          dummyResultReg = helperLink->buildDirectDispatch(node, &deps);
```

Comment:
If I remember correctly for the monent and monexit evaluator, is that we are in a unique situation here where we would have multiple OOL (In common case, we would have two OOL, One for cache look-up - which when failed calls helper, and another would also contain simple helper call. Reason for this kind of duplication was that we could not have jump from OOL to another OOL and entry point to OOL should be only one. 

for that, we passed in the &deps here to buildDirectDispatch as it will allocate / generate dependency for the Helper call which we should attach to the right position. I would take extra care of how things are done for OOL / helper call to avoid nasty bug which would be difficult to investigate.